contracts/forwarder/Forwarder.sol

```diff
@@ -85,13 +89,32 @@ contract Forwarder is IForwarder {
         registerRequestTypeInternal(requestType);
     }
 
+    function registerDomainSeparator(string calldata typePrefix, bytes memory dataPrefix) external override {
+        bytes memory domainType = abi.encodePacked(EIP712_DOMAIN_PREFIX, typePrefix, EIP712_DOMAIN_SUFFIX );
+        uint256 chainId;
+        /* solhint-disable-next-line no-inline-assembly */
+        assembly { chainId := chainid() }
+
+        bytes memory domainValue = abi.encodePacked(
+            keccak256(domainType),
+            dataPrefix,
+            chainId,
+            uint256(uint160(address(this))));
+
+        bytes32 domainHash = keccak256(domainValue);
```

Comment:
`domainSeparatorHash` was defined as EIP-712 hash of `EIP712Domain` structure. 
Here `domainHash` means a hash of type (as string) and some more data thrown in. 
I am not sure this is needed.

Reply:
see after the change.